src/client/dfuse/dfuse_pool.c

```diff
@@ -35,100 +34,56 @@ dfuse_pool_lookup(fuse_req_t req, struct dfuse_inode_entry *parent,
 	 */
 	D_ASSERT(parent->ie_stat.st_ino == parent->ie_dfs->dfs_ino);
 
-	D_ALLOC_PTR(dfp);
-	if (!dfp)
-		D_GOTO(err, rc = ENOMEM);
-
-	D_INIT_LIST_HEAD(&dfp->dfp_dfs_list);
-
 	/*
 	 * Dentry names with invalid uuids cannot possibly be added. In this
 	 * case, return the negative dentry with a timeout to prevent future
 	 * lookups.
 	 */
-	if (uuid_parse(name, dfp->dfp_pool) < 0) {
-		entry.entry_timeout = 60;
-		DFUSE_TRA_INFO(parent, "Invalid container uuid");
+	if (uuid_parse(name, pool) < 0) {
+		struct fuse_entry_param entry = {.entry_timeout = 60};
+
+		DFUSE_TRA_INFO(parent, "Invalid pool uuid");
 		DFUSE_REPLY_ENTRY(parent, req, entry);
-		D_FREE(dfp);
 		return;
 	}
 
-	D_MUTEX_LOCK(&fs_handle->dpi_info->di_lock);
-
-	d_list_for_each_entry(dfpi,
-			      &fs_handle->dpi_info->di_dfp_list,
-			      dfp_list) {
-		if (uuid_compare(dfp->dfp_pool, dfpi->dfp_pool) != 0)
-			continue;
-
-		d_list_for_each_entry(dfsi,
-				      &dfpi->dfp_dfs_list,
-				      dfs_list) {
-			{
-				if (uuid_is_null(dfsi->dfs_cont) != 1)
-					continue;
-
-				DFUSE_TRA_INFO(dfpi, "Found existing pool");
-
-				rc = dfuse_check_for_inode(fs_handle, dfsi,
-							   &ie);
-				D_ASSERT(rc == -DER_SUCCESS);
-
-				DFUSE_TRA_INFO(ie,
-					       "Reusing existing pool entry without reconnect");
-				entry.attr = ie->ie_stat;
-				entry.generation = 1;
-				entry.ino = entry.attr.st_ino;
-				DFUSE_REPLY_ENTRY(ie, req, entry);
-				D_MUTEX_UNLOCK(&fs_handle->dpi_info->di_lock);
-				D_FREE(dfp);
-				return;
-			}
-		}
-	}
+	DFUSE_TRA_DEBUG(parent, "Lookup of "DF_UUID,
+			DP_UUID(pool));
 
-	D_ALLOC_PTR(dfs);
-	if (!dfs)
-		D_GOTO(err_unlock, rc = ENOMEM);
+	rc = dfuse_pool_open(fs_handle, &pool, &dfp);
+	if (rc != 0)
+		goto err;
 
-	dfuse_dfs_init(dfs, parent->ie_dfs);
+	rc = dfuse_cont_open(fs_handle, dfp, &cont, &dfc);
+	if (rc != 0)
+		goto err;
 
-	d_list_add(&dfs->dfs_list, &dfp->dfp_dfs_list);
-	dfs->dfs_dfp = dfp;
+	/* Drop the reference on the pool */
+	d_hash_rec_decref(&fs_handle->dpi_pool_table, &dfp->dfp_entry);
 
-	DFUSE_TRA_UP(dfp, parent->ie_dfs->dfs_dfp, "dfp");
+	rlink = d_hash_rec_find(&fs_handle->dpi_iet,
+				&dfc->dfs_ino,
+				sizeof(dfc->dfs_ino));
+	if (rlink) {
+		struct fuse_entry_param	entry = {0};
```

Comment:
Similar to the container code, if we don't find the pool, should we be inserting it below?